test/src/unittests/rhythm/test_rhythmextractor2013.py

```diff
@@ -0,0 +1,230 @@
+        return(self._pulseTrain(bpm=140, sr=44100., offset=.1, dur=10))
+
+    def _createPulseTrainCombo(self):
+        # Define impulse train at 90bpm no offset
+        impulseTrain90 = self._pulseTrain(bpm=90., sr=44100., offset=0., dur=20.)
+        # Define impulse train at 140bpm no offset
+        impulseTrain140 = self._pulseTrain(bpm=140., sr=44100., offset=.1, dur=10.)
+        # Define impulse train at 200bpm with offset
+        impulseTrain200 = self._pulseTrain(bpm=200., sr=44100., offset=.2, dur=10.)
+        # Define impulse train at 200bpm with no offset
+        impulseTrain200b = self._pulseTrain(bpm=200., sr=44100., offset=0., dur=25.)
+        # Define Combination impulse train at 140-90-200-200 bpm
+        return(impulseTrain90 + impulseTrain140 + impulseTrain200 + impulseTrain200b)
+    
+    def testImpulseTrainMultifeature(self):
+        # Define expected values
+        impulseTrain140=self._createPulseTrain140()
+        expectedBpm = 140
+        expectedTicks = [i/44100. for i in range(len(impulseTrain140)) if impulseTrain140[i]!= 0]
+        expectedConfidence =3.8
+        #print("\nExpected Ticks Multifeature\n")
+        #print(expectedTicks)
+        # Test Multifeature with impulseTrain140
+        result = self._runInstance(impulseTrain140,method="multifeature")
+        # Check bpm
+        self.assertAlmostEqual(result[0], expectedBpm, 1e-2)
+        # Check ticks
+        self._assertVectorWithinVector(result[1], expectedTicks, 0.2)
+        #print("\nActual Ticks Multifeature\n")
+        #print(result[1])
+        # Check confidence
+        self.assertAlmostEqual(result[2], expectedConfidence,1.0)
+        # Check estimated bpm
+        for i in range(len(result[3])):
+            self.assertAlmostEqual(result[3][i], expectedBpm, 1.)
+        # Check bpm intervals
+        for i in range(len(result[4])):
+            self.assertAlmostEqual(result[4][i], 60./expectedBpm, 0.2)
+        # Test Multifeature with this combo impulseTrain
+        result = self._runInstance(self._createPulseTrainCombo(),method="multifeature")
+        # Define expected values
+        impulseTrainCombo= self._createPulseTrainCombo()
+        expectedBpm = 117 
+        expectedTicks = [i/44100. for i in range(len(impulseTrainCombo)) if impulseTrainCombo[i]!= 0]
+        expectedConfidence = 0.0
+        #print("\nExpected Ticks Combo Multifeature\n")
+        #print(expectedTicks)
+        # Check bpm
+        self.assertAlmostEqual(result[0], expectedBpm, .5)
+        # Check ticks
+        self._assertVectorWithinVector(result[1], expectedTicks,0.5 )
+        # Check confidence
+        expectedConfidence = 3.0
+        self.assertAlmostEqual(result[2], expectedConfidence, 1.0)
+        # Check estimate bpm
+        for i in range(len(result[3])):
+            self.assertAlmostEqual(result[3][i], expectedBpm, 5.0)
+        # bpm intervals: we may need to take into account also multiples of 90,
+        # 140 and 200.
```

Comment:
Simplify (remove redundancy in comments, better separation of different sub-tests  with empty lines)